src/module/loader/core-module-loader.h

```diff
@@ -24,9 +24,10 @@ namespace module {
 
         /// <summary> It loads a core module. </summary>
         /// <param name="name"> Core module name. </param>
+        /// <param name="arg"> Argument for loading the file. Passed through as arg1 from require. </param>
```

Comment:
It is node compatible, but not vice-versa. Napa module system is implemented differently from Node's, node's logic is mostly in JavaScript, thus Module class was introduced to facilitate, and all Napa implementation is from C++. 

This form of `require` can be seen as a private API, like Node's Module._compile etc. User cannot see it due to the node.d.ts they use.